File: Day-Four/python/Credit-Card-Validator/credit_card_validator_test.py
from unittest import TestCase

from credit_card_validator import *

class TestThatcredit_card_validatorWorks(TestCase):

    def test_that_get_card_type_returns_Visacard_if_card_starstWith_4(self):

        actual = get_card_type("4388576018402626")
        self.assertTrue(actual)      
    
    def test_that_get_card_type_returns_MasterCard_if_card_starstWith_5(self):

        actual = get_card_type("5388576018402626")
        self.assertTrue(actual)          

    def test_that_get_card_type_returns_American_Express_Card_if_card_starstWith_37(self):

        actual = get_card_type("378576018402626")
        self.assertEqual(actual, "American Express Card")          

    def test_that_get_card_type_return_invalid_card_if_card_starstWith_32(self):

        actual = get_card_type("328576018402626")
        self.assertEqual(actual, "Invalid card")

    def test_that_get_card_type_returns_invalidCard_if_card_starstWith_any_other_number(self):

        actual = get_card_type("778576018402626")
        self.assertEqual(actual, "Invalid card")       

    def test_that_valid_card_number_returns_valid(self):
        
        actual = get_card_validity("4388576018410707")
        self.assertEqual(actual, "Valid")   

File: Day-Four/javascript/Credit-Card-Validator/CreditCardValidator.test.js
const { getCardType, getCardLength, isValid } = require("./CreditCardValidator")

test("Test that cardNumber starting with 4 returns visa card", ()=> {

    let actual = getCardType(4388576018402626);
    let expected = "Visa Card";
    
    expect(actual).toEqual(expected);
})

test("Test that cardNumber starting with 5 returns MasterCard", ()=> {

    let actual = getCardType(5388576018402626);
    let expected = "MasterCard";
    
    expect(actual).toEqual(expected);
})

test("Test that cardNumber starting with 37 returns Amercan Express Card", ()=> {

    let actual = getCardType(3788576018402626);
    let expected = "American Express Card";
    
    expect(actual).toEqual(expected);
})

test("Test that cardNumber starting with 6 returns Discover Card", ()=> {

    let actual = getCardType(6388576018402626);
    let expected = "Discover Card";
    
    expect(actual).toEqual(expected);
})

test("Test that cardNumber starting with any other number returns Invslid Card", ()=> {

    let actual = getCardType(2388576018402626);
    let expected = "Invalid card";
    
    expect(actual).toEqual(expected);
})

test("Test that getCardLenngth Gets cardlength ", ()=> {

    let actual = getCardLength(4388576018402626);
    
    expect(actual).toEqual(16)
})

test("Test that valid ccard number returns 'Valid'", ()=> {

    let actual = isValid(4388576018410707);
    let expected = "Valid";
    
    expect(actual).toEqual(expected);
})

test("Test that invalid ccard number returns 'Invalid'", ()=> {

    let actual = isValid(4388576018402626);
    let expected = "Invalid";
    
    expect(actual).toEqual(expected);
})


